src/components/menus/ViewMenu.svelte

```diff
@@ -2,50 +2,75 @@
 
 <script lang="ts">
   import ChevronDownIcon from '@nasa-jpl/stellar/icons/chevron_down.svg?component';
-  import PlusIcon from 'bootstrap-icons/icons/plus.svg?component';
-  import SaveIcon from 'bootstrap-icons/icons/save.svg?component';
+  import ViewGridIcon from '@nasa-jpl/stellar/icons/view_grid_filled.svg?component';
+  import { createEventDispatcher } from 'svelte';
   import { user as userStore } from '../../stores/app';
-  import { view } from '../../stores/views';
-  import effects from '../../utilities/effects';
+  import { view, viewIsModified } from '../../stores/views';
+  import { showSavedViewsModal } from '../../utilities/modal';
+  import { Status } from '../../utilities/status';
+  import NavButton from '../app/NavButton.svelte';
   import Menu from './Menu.svelte';
   import MenuItem from './MenuItem.svelte';
 
+  const defaultViewName = 'Default View';
+  const dispatch = createEventDispatcher();
+
   let saveViewDisabled: boolean = true;
   let viewMenu: Menu;
 
-  $: saveViewDisabled = $view?.name === '' || $view?.owner !== $userStore?.id;
+  $: saveViewDisabled = $view?.name === '' || $view?.owner !== $userStore?.id || !$viewIsModified;
+
+  function saveAsView() {
+    if ($view) {
+      dispatch('create-view', { definition: $view.definition, owner: $userStore?.id });
+    }
+  }
 
-  function createView() {
+  function editView() {
     if ($view) {
-      effects.createView($userStore?.id, $view.definition);
+      dispatch('edit-view', { definition: $view.definition, owner: $userStore?.id });
     }
   }
 
   function saveView() {
     if ($view && $view.owner === $userStore?.id && !saveViewDisabled) {
-      effects.updateView($view.id, { definition: $view.definition, name: $view.name });
+      dispatch('save-view', { definition: $view.definition, id: $view.id, name: $view.name });
+    }
+  }
+
+  function resetView() {
+    if ($view) {
+      dispatch('reset-view');
     }
   }
 </script>
 
 <!-- svelte-ignore a11y-click-events-have-key-events -->
-<div class="view-menu st-typography-body" on:click|stopPropagation={() => viewMenu.toggle()}>
-  <ChevronDownIcon />
-
-  <Menu bind:this={viewMenu} offset={[-93, -5]}>
-    <MenuItem on:click={createView}>
-      <PlusIcon />
-      Create View
-    </MenuItem>
-    <MenuItem disabled={saveViewDisabled} on:click={saveView}>
-      <SaveIcon />
-      Save View
-    </MenuItem>
-  </Menu>
-</div>
+<NavButton
+  status={$viewIsModified ? Status.Modified : null}
+  title={$view?.name ?? defaultViewName}
+  on:click={() => viewMenu.toggle()}
+>
+  <ViewGridIcon />
```

Comment:
Is this 20px in Stellar? Think we'd want it to be 16px in the nav at least?

Reply:
Nice catch! I'm not sure if it should be 20px in stellar, but 20px is the set size when exporting from the accompanying figma so I didn't think to resize it since none of the other icons are resized. I've specified a height for now, but it should be removed once we update the overall UI to match the design.